model/riscv_insts_zvkned.sail

```diff
@@ -0,0 +1,550 @@
+/*
+ * Vector Cryptography Extension - NIST Suite: Vecttor AES Block Cipher
+ * ----------------------------------------------------------------------
+ */
+
+/*
+ * Helper functions.
+ * ----------------------------------------------------------------------
+ */
+
+val	 zvk_check_elements : (int, int, int, int) -> bool
+function zvk_check_elements(VLEN, num_elem, LMUL, SEW) = {
+  ((unsigned(vl)%num_elem) != 0) | ((unsigned(vstart)%num_elem) != 0) | (LMUL*VLEN) < (num_elem*SEW)
```

Comment:
the last term looks suspicious, is it trying to implement `EGW <= VLEN * LMUL` / `EGS <= VLMAX` ?
If so I do not think this is correct, for example I don't understand how the EGS / EGW is injected. Could a comment be added to explain which checks this function is supposed to perform ?

Reply:
This is implementing the following found in the specification, Version v0.4.2, 03 March, 2023.
```
if( ((vl%EGS)<>0) | ((vstart%EGS)<>0) | (LMUL*VLEN < EGW))  then {
      handle_illegal();  // illegal instruction exception
      RETIRE_SUCCESS
    } else {
    . . . 
```

Reply:
How is `EGW` derived from `num_elen * SEW` ?

Reply:
From the Vector Crypto Spec we have,
```
• Element Group Width (EGW) - total number of bits in an element group 
• Effective Element Width (EEW) - number of bits in each element
• Element Group Size (EGS) - number of elements in an element group
For all of the vector crypto instructions in this specification, EEW=SEW.
```

So here we're doing with `EGS * EEW`:
* num_elem = EGS
* SEW = EEW

Reply:
Sorry but I am still missing something, it looks like `num_elem` comes from `get_num_elem` and I do not get how `EGS` appears here:
```
$ git grep -A8 "fun.*get_num_elem" *.sail
model/riscv_vext_regs.sail:function get_num_elem(LMUL_pow, SEW) = {
model/riscv_vext_regs.sail-  let VLEN = int_power(2, get_vlen_pow());
model/riscv_vext_regs.sail-  let LMUL_pow_reg = if LMUL_pow < 0 then 0 else LMUL_pow;
model/riscv_vext_regs.sail-  /* Ignore lmul < 1 so that the entire vreg is read, allowing all masking to
model/riscv_vext_regs.sail-   * be handled in init_masked_result */
model/riscv_vext_regs.sail-  let num_elem = int_power(2, LMUL_pow_reg) * VLEN / SEW;
model/riscv_vext_regs.sail-  assert(num_elem > 0);
model/riscv_vext_regs.sail-  num_elem
model/riscv_vext_regs.sail-}
```

Reply:
I'll analyse this SAIL internal first and reach back. I'm positive `get_num_elem` returns the correct value, but a double-check is not a problem.